scapy/layers/l2.py

```diff
@@ -143,20 +167,23 @@ def i2h(self, pkt, x):
                 else:
                     x = "ff:ff:ff:ff:ff:ff"
                     warning("Mac address to reach destination not found. Using broadcast.")  # noqa: E501
-        return MACField.i2h(self, pkt, x)
+        return super(DestMACField, self).i2h(pkt, x)
 
     def i2m(self, pkt, x):
+        # type: (Optional[Ether], Optional[str]) -> bytes
         return MACField.i2m(self, pkt, self.i2h(pkt, x))
 
 
 class SourceMACField(MACField):
     __slots__ = ["getif"]
 
     def __init__(self, name, getif=None):
+        # type: (str, Optional[Any]) -> None
         MACField.__init__(self, name, None)
         self.getif = (lambda pkt: pkt.route()[0]) if getif is None else getif
 
     def i2h(self, pkt, x):
+        # type: (Optional[Ether], Optional[str]) -> str
```

Comment:
Technically, `SourceMACField` is used in other layers that `Ether`. You can probably just use `Packet`